pkg/daemon/ceph/osd/agent.go

```diff
@@ -142,26 +145,32 @@ func (a *OsdAgent) removeDirs(context *clusterd.Context, removedDirs map[string]
 	return nil
 }
 
-func (a *OsdAgent) configureDevices(context *clusterd.Context, devices *DeviceOsdMapping) ([]oposd.OSDInfo, error) {
-	var osds []oposd.OSDInfo
-	if devices == nil || len(devices.Entries) == 0 {
+func (a *OsdAgent) configureAllDevices(context *clusterd.Context, devices *DeviceOsdMapping) ([]oposd.OSDInfo, error) {
+
+	// prepare the OSDs with ceph-volume
+	osds, configured, err := a.configureDevices(context, devices)
+	if err != nil {
+		return nil, fmt.Errorf("failed to configure devices with ceph-volume. %+v", err)
+	}
+
+	if devices == nil || len(devices.Entries) == 0 || configured {
 		return osds, nil
 	}
 
+	// If ceph-volume is not supported, go ahead and configure the osds natively with rook
```

Comment:
I think we should fail here, I don't think we want to have more OSDs created this way if ceph-volume is not available then there is something wrong somewhere.
IMHO ceph-volume presence is a must-have.

Reply:
I believe we can enforce this once we have images for 12.2.10 and 13.2.3. In that case, we could add a check for the ceph version in the crd and fail much earlier than this. But I think we'll need to wait for the v0.9.1 release to enforce this since we haven't been able to get those images yet for testing.

Reply:
Sounds good as a followup.